lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake

```diff
@@ -0,0 +1,32 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '3.8' do
+      desc "Clear checksum type for on-demand repositories"
+      task :clear_checksum_type => %w(environment) do
+        User.current = User.anonymous_admin
+
+        repos = Katello::Repository.yum_type.where.not(checksum_type: nil, url: nil)
+        repos.find_each do |repo|
+          repo.transaction do
+            begin
+              if repo.on_demand?
+                repo.update_attribute(:checksum_type, nil)
+
+                if repo.find_distributor[:config]&.delete(:checksum_type)
+                  Katello.pulp_server.resources.repository.update_distributor(
+                    repo.pulp_id, repo.find_distributor[:id], repo.find_distributor[:config])
+                end
+              end
+
+              if repo.library_instance?
+                repo.update_attribute(:source_repo_checksum_type, repo.pulp_scratchpad_checksum_type)
```

Comment:
This did not work.  I'm not 100% sure why, but the object was never saved and even with update_attribute!, the record was not saved.  Changing to update_attributes!() seemed to work fine though? :confused:

Reply:
With these two changes it all seems to work well!